1_fetch/src/get_nwis_data.R

```diff
@@ -1,30 +1,43 @@
-download_nwis_data <- function(site_num, parameterCd, startDate, endDate){
+download_nwis_data <- function(site_num, parameterCd, startDate, endDate, fileout){
 
   data_out <- readNWISdata(sites = site_num, parameterCd = parameterCd, startDate = startDate, endDate = endDate, service="iv")
-
+  # data_out <- data.frame(data_out)
   # -- simulating a failure-prone web-service here, do not edit --
   set.seed(Sys.time())
   if (sample(c(T,F,F,F), 1)){
     stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
   }
-    
-  return(data_out)
   
+  ## CHANGE: added a write csv to get file 
+  write_csv(data_out, fileout)
+  
+  ## CHANGE: returns file path vs object previously
+  return(fileout)
+
 }
 
 ## Combine nwis datasets 
-combine_nwis_data <- function(...){
+combine_nwis_data <- function(file_path_folder){
```

Comment:
I see where you are going with this. 👍 

Having a directory as an input could save a lot of typing! But there are two problems here: 

1. The directory is not defined as a target. This means that targets would not be able to track whether changes have occurred to it, so changes to it would not be registered and won't trigger the pipeline to re-run. For an example of how directory targets work, see [How to depend on a directory for changes](https://github.com/msleckman/ds-pipelines-targets-2/issues/3#issuecomment-976133415) in #3.
2. Even if this directory were switched to a target, it wouldn't actually be best practice for targets. Directory targets are generally used when the directory already exists and files are already present in that directory before the pipeline is run (e.g., manually added files, the # of which or contents of which might change between pipeline runs, but not during pipeline runs). In the case when file targets are dynamically generated by the pipeline while it is running, it is better to have downstream targets depend explicitly on those file targets, so that all dependencies are clearly tracked. If a directory target is used when files in that directory are being dynamically generated by the pipeline, targets does not always catch changes to those files that would affect downstream targets. You'll learn in pipelines-III that targets has a better way to track large numbers of very similar targets.

Could you please change `combine_nwis_data` to explicitly take the `site_data_*csv` targets as input arguments instead?